### ISSUE TITLE
Infinite Loop on !shaped-update-character --all when some converting SKT monsters

### ISSUE THREAD

**Issue by Jimwaters** [symposion/roll20-shaped-scripts#389]
I have an SKT campaign I am bulk converting to the latest setup, and when I do I find some characters that cause an Infinite Loop crash.  [I have to then disable and reenable the script to get API restrated].

These characters convert fine if I copy in from my SKT game without character sheets, and convert singly using !shaped-update-character, or open and convert without using this command.  

My SKT campaign is v7 of sheet before conversion as I restored back to an old version,

So far the ones I have found are:-
Sprite
Imperator Uthor

It is odd that the issue only happens when use --all - and for just a few monsters.  Must be some issue with data - but perhaps it can be fixed?

**Comment by Jimwaters** [comment]
Did some further testing and seems to be one of those random issues.

Possible infinite loop detected, shutting down.

Occured on a character that was previously converted without issue - and indeed did not even need converting therefore!

**Comment by symposion** [comment]
Hey Jim
The infinite loop detector is fairly simple - it just checks that processes on the API server return in a timely fashion. If a script is occupied for too long, the server assumes that the author has done something stupid and put an infinite loop in there somewhere, and kills the sandbox. 

Unfortunately, in this instance, what's happening is that things are taking too long because you're doing a substantial operation on every character in a large campaign. There's really no way I can meaningfully optimise what I'm doing here. There's a chance that Kryx could make the sheetworkers a bit more efficient, but I doubt he's going to get that much more out of them. In reality, the performance bottleneck is most likely on Roll20's side. I've raised this with Riley, but his response is basically: we don't want you doing large batch operations on the API server anyway.
TL;DR: The Roll20 approved fix for this problem is to remove the feature from the script. So you're sh*t out of luck here.

**Comment by Jimwaters** [comment]
Thanks Lucian.

The convertion completed after a while - so this can be closed.
